Dockerfile

```diff
@@ -1,15 +1,20 @@
-FROM circleci/python:3.7-node
+FROM python:3.7
 USER root
 RUN mkdir -p /srv/code
 WORKDIR /srv/code
 #Copy code from current directory to /srv/code/ in image
 ADD . .
+
+RUN curl -fsSL https://deb.nodesource.com/setup_14.x | bash -
+
 # Install Python dependencies
-RUN pipenv install --system --deploy
-#Install static dependencies
+RUN apt-get update && apt-get install -y netcat git nodejs && \
+    pip install pipenv && \
+    pipenv install --system --deploy
+
 RUN npm i
 
-EXPOSE 80
+EXPOSE 8000 8080
 
-RUN npm rebuild node-sass
-RUN npm run build
\ No newline at end of file
+RUN npm rebuild node-sass && \
+    npm run build
```

Comment:
This works well as-is, but we might want to reorder the directives slightly so that the source code is added to the image after installing all the dependencies, but before doing a webpack build, that way any changes to the code don't automatically invalidate the other layers.

I did some experimenting, and this order seemed to work well:

```
FROM python:3.7
USER root
RUN mkdir -p /app
WORKDIR /app

# Install system dependencies
RUN curl -fsSL https://deb.nodesource.com/setup_14.x | bash - && \
    apt-get update && \
    apt-get install -y netcat git nodejs && \
    rm -rf /var/lib/apt/lists/*

# Install python dependencies
COPY ./Pipfile.lock ./Pipfile /app/
RUN pip install pipenv && \
    pipenv install --system --deploy && \
    rm Pipfile.lock Pipfile

# Install node dependencies
COPY ./package-lock.json ./package.json /app/
RUN npm i && \
    npm rebuild node-sass && \
    rm package-lock.json package.json  

# Copy app files
COPY . /app

# Build webpack assets 
RUN npm run build

# Use port 8000 for django server or 8080 for webpack dev server
EXPOSE 8000 8080
```

Note that `COPY` replaces `ADD` to copy the source code into the image. According to [docker best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy), `COPY` should be  preferred.